arangod/Aql/Ast.cpp

```diff
@@ -2280,6 +2280,12 @@ std::unordered_set<std::string> Ast::getReferencedAttributesForKeep(
         // do not descend further
         return false;
       }
+      if(checkForAttributeName){
```

Comment:
`RETURN other[0].first[0]`

will have an expression of the following form:

```
      - indexed access
          - attribute access: first
            - indexed access
              - reference: other
              - value: 0
          - value: 0
```

To me it looks like the indexed access hast some value for the index and another member of different type that provides information about the variable. I am not sure how you want me to use the NODE_TYPE_INDEXED_ACCESS  directly to get attribute names or referenced variables. For the rule above I end up within the indexed access and stop.

Reply:
Let's say the query is something like
```
COLLECT unused1 = ... INTO g1
COLLECT unused2 = ... INTO g2
RETURN g2[0].g1[0].foo
```
If there is the expression `g2[0].g1[0].foo`, then it would be safe to keep just the `foo` subattribute from `g1`.